nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestAutomaticIndexing.java

```diff
@@ -1075,4 +1082,120 @@ public void pathLevelFieldMustBeSeenAsKeyword() throws Exception {
         waitForCompletion();
         startTransaction();
     }
+
+    @Test
+    public void shouldIndexUpdatedRecord() throws Exception {
+        startTransaction();
+        DocumentModel doc = session.createDocumentModel("/", "mydoc", "File");
+        doc = session.createDocument(doc);
+        session.save();
+
+        TransactionHelper.commitOrRollbackTransaction();
+        waitForCompletion();
+        startTransaction();
+
+        // no record found in index
+        String nxql = "SELECT * FROM Document WHERE ecm:isRecord = 1";
+        DocumentModelList ret = ess.query(new NxQueryBuilder(session).nxql(nxql));
+        Assert.assertEquals(0, ret.totalSize());
+
+        // make the doc a record
+        session.makeRecord(doc.getRef());
+        session.save();
+
+        TransactionHelper.commitOrRollbackTransaction();
+        waitForCompletion();
+        assertNumberOfCommandProcessed(1); // update
+        startTransaction();
+
```

Comment:
i am wondering what if we use `transactionalFeature.nextTransaction();` in both cases (wait indexing after creation and updating) -> and we can adapt the code 
```
transactionalFeature.nextTransaction();
assertNumberOfCommandProcessed(1); // creation
........
transactionalFeature.nextTransaction();
assertNumberOfCommandProcessed(2); // + update
```

Reply:
Again that's copy/pasted code from other tests that work well, so no big incentive to change it here.